### ISSUE TITLE
Audio import problem - right single quote in file name causes audio import to fail with empty file.

### ISSUE THREAD

**Issue by Chester-Graham** [ElvishArtisan/rivendell#640]
I just had a group of 60 .flac files imported using dropbox into Rivendell, two of them failed to import. All were removed from the folder as if they were successful. The two files were in the cart with proper name, but no audio.

The log reported 
Jan 24 16:44:20:  Importing file "54_The_Church_God's_Storehouse.flac" [54 The Church God's Storehouse] to cart 009450 ... Jan 24 16:44:24: done.
Jan 24 16:44:24:  Deleted file "54_The_Church_God's_Storehouse.flac"

HOWEVER - - - when I copied the lines into windows they came across like this with the "apostrope" actually being a "right single quote mark".
Jan 24 16:44:20:  Importing file "54_The_Church_God\u2019s_Storehouse.flac" [54 The Church God\u2019s Storehouse] to cart 009450 ... Jan 24 16:44:24: done.
Jan 24 16:44:24:  Deleted file "54_The_Church_God\u2019s_Storehouse.flac"

If I remove the "apostrophe" from the file name they import correctly. There doesn't seem to be a way of differentiating from the "apostrope" and the "right single quote". They both look the same in the file name.

Dropbox configuration from log file 
_____________
Jan 03 23:23:38: rdimport startedJan 03 23:23:38: RDImport v3.5.0
Jan 03 23:23:38:  Force to Mono is OFF
Jan 03 23:23:38:  Normalization level = -1 dB
Jan 03 23:23:38:  AutoTrim level = -40 dB
Jan 03 23:23:38:  Destination cart is 009449
Jan 03 23:23:38:  Single cart mode is ON
Jan 03 23:23:38:  Delete source mode is ON
Jan 03 23:23:38:  Delete cuts mode is OFF
Jan 03 23:23:38:  DropBox mode is ON
Jan 03 23:23:38:  Using metadata pattern: %i.
Jan 03 23:23:38:  Start Date Offset = 0 days
Jan 03 23:23:38:  End Date Offset = 0 days
Jan 03 23:23:38:  Broken format workarounds are ENABLED
Jan 03 23:23:38:  E-mail reporting is DISABLED
Jan 03 23:23:38:  Import Create Dates mode is OFF
Jan 03 23:23:38:  Persistent DropBox ID = 19
Jan 03 23:23:38:  Files to process:
Jan 03 23:23:38:    "/home/rd/music_import/Specific Cart 2/*"
Jan 03 23:23:38: rdimport startedJan 03 23:23:38: RDImport v3.5.0
Jan 03 23:23:38: rdimport startedJan 03 23:23:38:  Force to Mono is OFF
Jan 03 23:23:38: RDImport v3.5.0
Jan 03 23:23:38:  Normalization level = -1 dB
Jan 03 23:23:38:  Force to Mono is OFF
Jan 03 23:23:38:  AutoTrim level = -40 dB
Jan 03 23:23:38:  Normalization level = -1 dB
Jan 03 23:23:38:  Destination cart is 009448
Jan 03 23:23:38:  AutoTrim level = -40 dB
Jan 03 23:23:38:  Single cart mode is ON
Jan 03 23:23:38:  Delete source mode is ON
Jan 03 23:23:38:  Destination cart is 009450
Jan 03 23:23:38:  Delete cuts mode is OFF
Jan 03 23:23:38:  DropBox mode is ON
Jan 03 23:23:38:  Single cart mode is ON
Jan 03 23:23:38:  Using metadata pattern: %i.
Jan 03 23:23:38:  Start Date Offset = 0 days
Jan 03 23:23:38:  Delete source mode is ON
Jan 03 23:23:38:  End Date Offset = 0 days
Jan 03 23:23:38:  Delete cuts mode is OFF
Jan 03 23:23:38:  Broken format workarounds are ENABLED
Jan 03 23:23:38:  E-mail reporting is DISABLED
Jan 03 23:23:38:  Import Create Dates mode is OFF
Jan 03 23:23:38:  DropBox mode is ON
Jan 03 23:23:38:  Persistent DropBox ID = 7
Jan 03 23:23:38:  Files to process:
Jan 03 23:23:38:  Using metadata pattern: %i.
Jan 03 23:23:38:    "/home/rd/music_import/Specific Cart 1/*"
Jan 03 23:23:38:  Start Date Offset = 0 days
Jan 03 23:23:38:  End Date Offset = 0 days
Jan 03 23:23:38:  Broken format workarounds are ENABLED
Jan 03 23:23:38:  E-mail reporting is DISABLED
Jan 03 23:23:38:  Import Create Dates mode is OFF
Jan 03 23:23:38:  Persistent DropBox ID = 20
Jan 03 23:23:38:  Files to process:
Jan 03 23:23:38:    "/home/rd/music_import/Specific Cart 3/*"
Jan 03 23:23:38: rdimport startedJan 03 23:23:38: RDImport v3.5.0
Jan 03 23:23:38:  Force to Mono is OFF
Jan 03 23:23:38:  Normalization level = -1 dB
Jan 03 23:23:38:  AutoTrim is OFF
Jan 03 23:23:38:  Destination cart is AUTO
Jan 03 23:23:38:  Single cart mode is OFF
Jan 03 23:23:38:  Delete source mode is ON
Jan 03 23:23:38:  Delete cuts mode is OFF
Jan 03 23:23:38:  DropBox mode is ON
Jan 03 23:23:38:  Adding Scheduler Code(s):
Jan 03 23:23:38:  Z
Jan 03 23:23:38:  Using metadata pattern: %i#%t#%a.
Jan 03 23:23:38:  Start Date Offset = 0 days
Jan 03 23:23:38:  End Date Offset = 0 days
Jan 03 23:23:38:  Broken format workarounds are ENABLED
Jan 03 23:23:38:  E-mail reporting is DISABLED
Jan 03 23:23:38:  Import Create Dates mode is OFF
Jan 03 23:23:38:  Persistent DropBox ID = 18
Jan 03 23:23:38:  Files to process:
Jan 03 23:23:38:    "/home/rd/music_import/*"
Jan 03 23:24:08:  Importing file "01 - Our Day in Light of Bible Prophecy (Dimensions of  Prophecy Kenneth Cox).flac" [01. Our Day in Light of Bible Prophecy (Dimensions of  Prophecy Kenneth Cox)] to cart 009448 ... Jan 03 23:26:11: done.
Jan 03 23:26:11:  Deleted file "01 - Our Day in Light of Bible Prophecy (Dimensions of  Prophecy Kenneth Cox).flac"
Jan 03 23:26:11:  Importing file "02 - United States in Bible Prophecy (Dimensions of Prophecy Kenneth Cox).flac" [02. United States in Bible Prophecy (Dimensions of Prophecy Kenneth Cox)] to cart 009448 ... Jan 03 23:28:41: done.
Jan 03 23:28:41:  Deleted file "02 - United States in Bible Prophecy (Dimensions of Prophecy Kenneth Cox).flac"
Jan 03 23:28:41:  Importing file "03 - Signs of Christ's Return (Dimensions of Prophecy Kenneth Cox).flac" [03 - Signs of Christ's Return (Dimensions of Prophecy Kenneth Cox)] to cart 009448 ... Jan 03 23:30:41: done.
Jan 03 23:30:41:  Deleted file "03 - Signs of Christ's Return (Dimensions of Prophecy Kenneth Cox).flac"
Jan 03 23:30:41:  Importing file "04 - Reunion Day in Heaven (Dimensions of Prophecy Kenneth Cox).flac" [04 - Reunion Day in Heaven (Dimensions of Prophecy Kenneth Cox)] to cart 009448 ... Jan 03 23:32:58: done.
Jan 03 23:32:58:  Deleted file "04 - Reunion Day in Heaven (Dimensions of Prophecy Kenneth Cox).flac"
Jan 03 23:32:58:  Importing file "05 - Short Beds and Narrow Covers (Dimensions of Prophecy Kenneth Cox).flac" [05 - Short Beds and Narrow Covers (Dimensions of Prophecy Kenneth Cox)] to cart 009448 ... Jan 03 23:34:53: done.
Jan 03 23:34:53:  Deleted file "05 - Short Beds and Narrow Covers (Dimensions of Prophecy Kenneth Cox).flac"
Jan 03 23:34:53:  Importing file "06 - The Abomination of Desolation (Dimensions of Prophecy Kenneth Cox).flac" [06 - The Abomination of Desolation (Dimensions of Prophecy Kenneth Cox)] to cart 009448 ... Jan 03 23:37:16: done.
Jan 03 23:37:16:  Deleted file "06 - The Abomination of Desolation (Dimensions of Prophecy Kenneth Cox).flac"
Jan 03 23:37:16:  Importing file "07 - The Book of Life on Judgement Day (Dimensions of Prophecy Kenneth Cox).flac" [07 - The Book of Life on Judgement Day (Dimensions of Prophecy Kenneth Cox)] to cart 009448 ... Jan 03 23:39:49: done.
Jan 03 23:39:49:  Deleted file "07 - The Book of Life on Judgement Day (Dimensions of Prophecy Kenneth Cox).flac"
Jan 03 23:39:49:  Importing file "08 - How to Know God's Will (Dimensions of Prophecy Kenneth Cox).flac" [08 - How to Know God's Will (Dimensions of Prophecy Kenneth Cox)] to cart 009448 ... Jan 03 23:42:01: done.
Jan 03 23:42:01:  Deleted file "08 - How to Know God's Will (Dimensions of Prophecy Kenneth Cox).flac"

**Comment by ElvishArtisan** [comment]
Would you by any chance have a sample of one of these files with the right apostrophe available for testing?

**Comment by Chester-Graham** [comment]
Here is an example

**Comment by Chester-Graham** [comment]
Not sure if the file actually got thru, here it is again . . . it is actually a .flac file, but had to tag it with .mp4 to upload. just remove the .mp4

https://user-images.githubusercontent.com/63725659/113789478-2cfbca80-970d-11eb-95d2-5e73529d4ede.mp4

**Comment by ElvishArtisan** [comment]
Got it, thanks.

**Comment by ElvishArtisan** [comment]
Unfortunately, GitHub manged the name of the file to some huge hexidecimal mess. However, after renaming it to "54_The_Church_God's_Storehouse.flac", it imports perfectly via dropbox and plays without problem, with the apostrophe being processed correctly through a metadata pattern rule.

I'm not sure what happened with those original two files, but I suspect that they may have been corrupt somehow. I'm not seeing any problems with Rivendell processing RIGHT_SINGLE_QUOTE_MARK Unicode characters.

**Comment by Chester-Graham** [comment]
Weird, I did a manual import of it on my system and it imported as a zero
length file. Will check again.

**Comment by ElvishArtisan** [comment]
'Kay. I'm going to close this issue in the interim. If you do come up with the original files, or a way to consistently goose the problem, feel free to reopen!

**Comment by Chester-Graham** [comment]
I have done further testing, and the plot thickens. The original file was created by Izotope RX8 as a flac file with highest compression. This original flac file will not import into Rivendell the "apostrophe"(unicode 2019) in the file name.
If I replace unicode 2019 with a regular apostrophe (unicode 0027) in the file name, the .flac file will import.
I Do NOT see this same behavior if the file is a .mp3. In that case it imports just fine with either character.

I have located the original problem file, and also a second file with the same issue. They are too big to send thru this message board at 245,534 KB. I would be happy to forward the original file to you, but need a different way of doing so.

**Comment by Chester-Graham** [comment]
I have uploaded them to dropbox, link is https://www.dropbox.com/sh/3dcui7qj1k4c12r/AAAPwtf3nkRJ9KNuDEqYIakXa?dl=0

**Comment by ElvishArtisan** [comment]
Got it, thanks.

**Comment by ElvishArtisan** [comment]
Confirmed here.

**Comment by ElvishArtisan** [comment]
Fixed in 'master'.

**Comment by Chester-Graham** [comment]
Thanks

**Comment by Chester-Graham** [comment]
I have another batch of flac files that will not import either by dropbox or directly in rdlibrary. There are 15 in total. I have uploaded a representative to the dropbox folder that we used before . . . https://www.dropbox.com/sh/3dcui7qj1k4c12r/AAAPwtf3nkRJ9KNuDEqYIakXa?dl=0
Not sure if the problem that you fixed also fixed these, but letting you know in case there is another problem with flac importing.
